aineko/extras/connectors/rest_poller.py

```diff
@@ -0,0 +1,260 @@
+
+from aineko import AbstractNode
+
+
+class ParamsRESTPoller(BaseModel):
+    """Connector params for REST model."""
+
+    timeout: int = 10
+    url: str
+    headers: Optional[Dict[str, Any]] = None
+    data: Optional[Dict[str, Any]] = None
+    poll_interval: float = 5.0
+    max_retries: int = -1
+    metadata: Optional[Dict[str, Any]] = None
+    retry_sleep: float = 5
+    success_codes: List[int] = [
+        200,
+        201,
+        202,
+        203,
+        204,
+        205,
+        206,
+        207,
+        208,
+        226,
+    ]
+
+    @field_validator("url")
+    @classmethod
+    def supported_url(cls, url: str) -> str:
+        """Validates that the url is a valid HTTP or HTTPS URL."""
+        if not (url.startswith("https://") or url.startswith("http://")):
+            raise ValueError(
+                "Invalid url provided to HTTPS params. "
+                'Expected url to start with "https://". '
+                f"Provided url was: {url}"
+            )
+        return url
+
+
+# pylint: disable=anomalous-backslash-in-string
+class RESTPoller(AbstractNode):
+    """Connects to an REST endpoint via HTTP or HTTPS and polls.
+
+    This node is a wrapper around the
+    [requests](
+        https://docs.python-requests.org/en/master/
+        ){:target="\_blank"} library.
+
+    `node_params` should be a dictionary with the following keys:
+
+            url: The REST URL to connect to
+            headers (optional): A dictionary of headers to send to the REST
+                endpoint. Defaults to None.
+            data (optional): A dictionary of data to send to the REST endpoint.
+                Defaults to None.
+            poll_interval (optional): The number of seconds to wait between
+                polls. Defaults to 5.
+            max_retries (optional): The maximum number of times to retry
+                connecting to the REST endpoint. Defaults to -1.
+            retry_sleep (optional): The number of seconds to wait between
+                retries. Defaults to 5.
+            metadata (optional): A dictionary of metadata to attach to outgoing
+                messages. Defaults to None.
+            success_codes (optional): A list of HTTP status codes that indicate
+                success. Defaults to
+                    [200, 201, 202, 203, 204, 205, 206, 207, 208, 226].
+
+    Secrets can be injected (from environment) into the `url`, `headers`, and
+    `data` fields by passing a string with the following format:
+    `{$SECRET_NAME}`. For example, if you have a secret named `SECRET_NAME`
+    that contains the value `SECRET_VALUE`, you can inject it into the url
+    field by passing `https://example.com?secret={$SECRET_NAME}` as the url.
+    The connector will then replace `{$SECRET_NAME}` with `SECRET_VALUE` before
+    connecting to the REST endpoint.
+
+    Example usage in pipeline.yml:
+    ```yaml title="pipeline.yml"
+    pipeline:
+      nodes:
+        RestPoller:
```

Comment:
The class is named `RESTPoller`. There is another incorrect reference a few lines below

Reply:
fixed